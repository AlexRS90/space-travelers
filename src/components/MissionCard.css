.card-wrapper {
    border: solid 2px brown;
    display:flex;
    justify-content: space-between;
    text-align: center;
    align-items: center;
}

.mission-desc {
    width: 70%;
    border-left: red 1px solid;
    border-right: red 1px solid;
}

.mission-action {
    width: 20%;
    text-align: center;
}
