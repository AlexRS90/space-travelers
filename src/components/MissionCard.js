import React from 'react';

const MissionCard = () => (
  <>
    <h1>Hello card</h1>
  </>
);

export default MissionCard;
