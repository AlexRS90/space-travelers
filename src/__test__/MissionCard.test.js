/* eslint-disable max-len */
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MissionCard from '../components/MissionCard';

afterEach(() => {
  cleanup();
});

const missions = [
  {
    mission_name: 'Thaicom',
    mission_id: 'EFDRT',
    reserved: true,
  },
  {
    mission_name: 'Thaicom',
    mission_id: 'HJUIY',
    reserved: false,
  },
  {
    mission_name: 'Falcon',
    mission_id: 'EFWERT',
    reserved: true,
  },
];

const joinTheMission = async () => {

};

const leaveMission = async () => {

};

describe('render the component to the dom', () => {
  test('Render mission card to the DOM', async () => {
    render(<MissionCard mission={missions} onClick={missions.reserved === true ? leaveMission : joinTheMission} />);
    const missionCardElement = await screen.getByTestId('mission-card');
    expect(missionCardElement).toBeInTheDocument();
  });

  test('matches the snapshot', () => {
    const tree = renderer.create(<MissionCard mission={missions} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
