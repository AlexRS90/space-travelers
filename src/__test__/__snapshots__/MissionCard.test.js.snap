// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the component to the dom matches the snapshot 1`] = `
<tbody>
  <tr
    data-testid="mission-card"
  >
    <td
      className="mission-title"
    />
    <td
      className="mission-desc"
    />
    <td
      className="join-mission"
    >
      <span
        className="me-3 span-top"
        style={
          Object {
            "backgroundColor": "#282828",
          }
        }
      >
        NOT A MEMBER
      </span>
    </td>
    <td
      className="member-status"
    >
      <button
        className="span-down"
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "border": "2px solid black",
            "color": "black",
          }
        }
        type="button"
      >
        Join Mission
      </button>
    </td>
  </tr>
</tbody>
`;
